### PR TITLE
Added migration guide for 1.15

### PR DESCRIPTION
**Pull request by mbrandonw**
This is a potential small issue that could come up with 1.15, and we've seen it brought up at least 2 or 3 times so far, so thought we could get a migration guide for it.

### CODE REVIEW

File: Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.15.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  .map { .response($0) }
+```
+
+There is also an upcoming Swift feature that will fix this. You can enable it in an SPM package
+by adding a `enableUpcomingFeature` to its Swift settings:
+
+```swift
+swiftSettings: [
+  .enableUpcomingFeature("InferSendableFromCaptures"),
+]),
```
Comment (JimRoepcke) [inline Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.15.md:41]: Sadly I found this wasn't effective in my codebase. We're on Xcode 15.4. Even with `-enable-experimental-feature InferSendableFromCaptures -enable-upcoming-feature InferSendableFromCaptures` in the `SwiftCompile` output in the build log, I still get that error.

**Review by stephencelis — APPROVED** [review]

File: Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.15.md (hunk 1 above)
Comment (mbrandonw) [reply Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.15.md:41]: Hi @JimRoepcke, I'm not able to reproduce this, but if you can provide a minimal project that demonstrates the problem we will look into more. For now I am going to merge this as-is.

Comment (r-plus) [reply Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.15.md:41]: Hi @JimRoepcke 
If you using local SwiftPackage, you should configure upcoming feature on Package.swift for SwiftPackage source code.
Configured `OTHER_SWIFT_FLAGS` in xcodeproj is not affect to SwiftPackage in my case.

Comment (JimRoepcke) [reply Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.15.md:41]: My mistake was I was using Xcode 15.4 thinking this feature flag worked in that toolchain. It requires Xcode 16!

**Pull request merged**